GA/HeaderFiles/ConstraintPUEO.h

```diff
@@ -34,6 +35,10 @@ bool ConstraintPUEO(float s, float h, float x_0,
   {
     intersect = true;
   }
+  else if( (-2*(y_0) + x_0 * (2*H/l + l/(2*H)))/(l/(2*H) - 2*H/l) > (x_0 - H) && (-2*(y_0) + x_0 * (2*H/l + l/(2*H)))/(l/(2*H) - 2*H/l) < x_0 )
```

Comment:
```suggestion
  else if((-2*(y_0) + x_0 * (2*H/l + l/(2*H)))/(l/(2*H) - 2*H/l) > (x_0 - H) &&
          (-2*(y_0) + x_0 * (2*H/l + l/(2*H)))/(l/(2*H) - 2*H/l) < x_0)
```
I would rewrite it like this to make it more readable.

Also since the first operand is the same, you could calculate it once at the beginning, assign it to a var with a descriptive name, and then use the var here twice, even though if you this it will have to calculate its value even when any of the previous conditions are true.

Adding a comment to explain what this condition is checking would also help.

Reply:
Sorry, I totally forgot about this. I'll write up an ELOG that explains it. In the mean time, I made a correction and wrote it more cleanly and with comments per your suggestion.